packages/jsActions/nanoflow-actions-native/src/geolocation/RequestLocationPermission.ts

```diff
@@ -14,11 +15,97 @@ import Geolocation, { GeolocationStatic } from "@react-native-community/geolocat
 export async function RequestLocationPermission(): Promise<boolean> {
     // BEGIN USER CODE
 
+    let rnGeolocation: GeolocationServiceStatic | GeolocationStatic;
+
     if (navigator && navigator.product === "ReactNative") {
-        if (!navigator.geolocation) {
-            (navigator.geolocation as GeolocationStatic) = Geolocation;
+        if (NativeModules.RNFusedLocation) {
+            const geolocationService = await import("react-native-geolocation-service");
+            rnGeolocation = geolocationService.default;
+        } else if (NativeModules.RNCGeolocation) {
+            rnGeolocation = Geolocation;
+        } else {
+            return Promise.reject(new Error("Geolocation module couldn't find"));
+        }
+    } else if (navigator && navigator.geolocation) {
+        rnGeolocation = navigator.geolocation;
+    } else {
+        return Promise.reject(new Error("Navigator couldn't find"));
+    }
+
+    const hasPermissionIOS = async (): Promise<boolean> => {
+        const openSetting = (): void => {
+            Linking.openSettings().catch(() => {
+                Alert.alert("Unable to open settings.");
+            });
+        };
+
+        if (!rnGeolocation?.requestAuthorization) {
+            return Promise.reject(new Error("requestAuthorization is undefined"));
```

Comment:
Can you come up with a more helpful end-user message? I don't think end-users need to know about `undefined`. 

Under what circumstances will `requestAuthorization` be `undefined`? Let the end-user know about this. Do they need to perform steps? Or is something missing?

Reply:
I fixed that. There is no reason to check `requestAuthorization` because it will never be `undefined`.